### PR TITLE
compelete react-state-3_journal-app-filter challenge

### PR DESCRIPTION
**Pull request by spacedoe**
# Context

This PR aims to tackle [The react-state-3_journal-app-filter challenge](https://github.com/spiced-academy/honey-web-dev/blob/main/sessions/react-state-3/challenges-react-state-3.md).

# Changes:

1. Made the `Tab` buttons clickable: 
- [x] `onClick` prop in `Tag` function declaration and passed it to the `button` tag. 

2. Informed the parent component `EntriesSection` when the buttons are clicked:
- [x] Added props `onShowAllEntries` and `onShowFavoriteEntries` to `EntriesSection` function declaration and passed them down as a prop to the `Tab` component via the prop `onClick`.

3. Stored the desired filter option in state:
- [x] In `App` created filter state with initial value "all".
- [x] Created two functions `handleShowAllEntries` and `onShowFavoriteEntries`  which set filter state to "all" or "favorites" respectively, passed them down to the `EntriesSection` component via the props `onShowAllEntries` and `onShowFavoriteEntries` 

4. Added filter logic: 
- [x] Created `favoriteEntries` array with the filtered values of  `entries` where `isFavorite` is `true`. 
- [x]  Passed the value of `entries` prop of the `EntriesSection` in the `App` conditionally with a ternary operator. 

5. Added Visual highlight for active filter:
- [x]  Passed `filter` prop to the `EntriesSection` in the `App` and  in `EntriesSection` function declaration.
- [x] Passed the value of the `isActive` prop of the `Badge` conditionally as per instructions. 

6. Adjusted the count of "all" and "favorite" entries:
- [x] Added props`allEntriesCount` and `favoriteEntriesCount` to the `EntriesSection` function declaration. 
- [x] Replaced the hard-coded numbers inside the `Tab` components with `allEntriesCount` and `favoriteEntriesCount` respectively. 
- [x] Added `allEntriesCount` and `favoriteEntriesCount` with calculated values using `.length` to the `EntriesSection` component in the `App`.

# Notes: 
there's a typo in README file on line 39: should be `filter` instead `showFavorites`

### CODE REVIEW

File: react/react-state-3/react-state-3_journal-app-filter/src/components/EntriesSection/index.js [hunk 1]
```diff
@@ -6,15 +6,23 @@ import Tab from "../Tab";
...
+  filter,
+  allEntriesCount,
+  favoriteEntriesCount,
+}) {
   return (
     <section className="entries-section">
       <Tabs>
-        <Tab active>
-          All Entries <Badge isActive>3</Badge>
+        <Tab active onClick={onShowAllEntries}>
```
Comment (spacedoe) [inline react/react-state-3/react-state-3_journal-app-filter/src/components/EntriesSection/index.js:21]: isActive={filter === "all'}

File: react/react-state-3/react-state-3_journal-app-filter/src/components/EntriesSection/index.js [hunk 2]
```diff
@@ -6,15 +6,23 @@ import Tab from "../Tab";
...
+          All Entries <Badge isActive={filter === "all"}>{allEntriesCount}</Badge>
         </Tab>
-        <Tab>
-          Favorites <Badge>1</Badge>
+        <Tab onClick={onShowFavoriteEntries}>
```
Comment (spacedoe) [inline react/react-state-3/react-state-3_journal-app-filter/src/components/EntriesSection/index.js:24]: isActive={filter === "favorites"}

File: react/react-state-3/react-state-3_journal-app-filter/src/components/EntriesSection/index.js (hunk 1 above)
Comment (reverbePrintemps) [reply react/react-state-3/react-state-3_journal-app-filter/src/components/EntriesSection/index.js:21]: I'm not sure what this comment is for? 🤔

File: react/react-state-3/react-state-3_journal-app-filter/src/components/EntriesSection/index.js (hunk 2 above)
Comment (reverbePrintemps) [reply react/react-state-3/react-state-3_journal-app-filter/src/components/EntriesSection/index.js:24]: Same here? 🤔

**Review by reverbePrintemps — APPROVED** [review]
I'm not sure why you left comments on your own code instead of changing your code to whatever your comments are stating but besides that, it's looking great! 🤘

**Comment by reverbePrintemps** [comment]
ps: absolutely love how explicit your PR description is. 😍 (just don't forget to tick the boxes for the things you have already done 😄)

File: react/react-state-3/react-state-3_journal-app-filter/src/components/EntriesSection/index.js (hunk 1 above)
Comment (spacedoe) [reply react/react-state-3/react-state-3_journal-app-filter/src/components/EntriesSection/index.js:21]: this comment means I need to add this code, as this is what's missing from the solution Jan showed us this morning.

**Comment by spacedoe** [comment]
Thx! Oh, i didn't know I could tick the boxes, thanks for letting me know.

**Pull request merged**